### ISSUE TITLE
"Secrets" on "Stages"

### ISSUE THREAD

**Issue by jjangga0214** [vercel/vercel#2741]
_TL;DR: Let us register a same 'secret' to multiple 'stages' with different values ._

_DISCLAIMER: This might be out of topic from now-cli, but I personally strongly feel this is a critical issue, therefore I post it here._

Hi,

According to the documentation(<https://zeit.co/docs/v2/serverless-functions/env-and-secrets/>), **secrets** are absolutely only identified by there name. However, I believe there are many cases we need separate environments (.e.g .env.development, .env.production, .env.qa ...). If we only depends on CI server, this is OK. But this becomes a problem when using 'Now For Github' app.

For example, let's say my API server needs this env var "DB_ENDPOINT".

```json
{
  "env": {
    "DB_ENDPOINT": "@db-endpoint"
  }
}
```

And I want to set "@db-endpoint" DIFFERENTLY when deploying by PR and master. On master, I want "@db-endpoint" to be a production database endpoint. On PR or other branches, test db endpoint.

So, I think an introduction of a concept like **stage** could be one of the possible solutions.

```bash
# registers secret "@db-endpoint" for 'prod' stage
now secrets add db-endpoint 'production_value' --stage 'prod'
# registers secret "@db-endpoint" for 'test' stage
now secrets add db-endpoint 'test_value' --stage 'test'
```

Therefore, we would be able to match the 'prod' stage with master, and 'test' with every other branch or PRs. Of course, users should be able to match them freely and name stages as they please.

So, in conclusion, fine-grained configuration by branch and environment is necessary in order to use now with 'Now For Github', which now thinks the right way of using its platform.

How do you feel?

**Comment by leo** [comment]
Thank you for creating this issue!

Please note that – in order for us to be able to handle each feature request with care – we have to consume all of them through a unified pipeline that makes it easier for us to prioritize, track and progress on the features our user base is interested in.

In turn, the [Issues tab](https://github.com/zeit/now/issues) on this repository is best only be used for reporting bugs, which we can immediately act on.

For well-explained feature requests (like the one you just posted), please contact us at [support@zeit.co](mailto:support@zeit.co), so that we can consider adding it to our roadmap.

In the case of your specific feature recommendation, you can rest assured that we're already tracking it on our roadmap and you can expect to see an update on our [Twitter account](https://twitter.com/zeithq) or [Blog](https://zeit.co/blog) soon.
